dimod/constrained.py

```diff
@@ -335,23 +333,62 @@ def add_constraint_from_comparison(self,
         Returns:
             Label of the added constraint.
 
-        Examples:
-            >>> from dimod import ConstrainedQuadraticModel, Integer
-            >>> i = Integer('i')
-            >>> cqm = ConstrainedQuadraticModel()
-            >>> cqm.add_constraint_from_comparison(i <= 3, label='Max i')
-            'Max i'
-            >>> print(cqm.constraints["Max i"].to_polystring())
-            i <= 3
+        Example:
 
-            Adding a constraint without copying the comparison's model requires
-            caution:
+            Encode a constraint
+
+            .. math::
+
+                x + y + xy <= 1
+
+            First create the relevant variables and the model.
+
+            >>> x, y = dimod.Binaries(['x', 'y'])
+            >>> cqm = dimod.ConstrainedQuadraticModel()
+
+            And add the constraint to the model
+
+            >>> cqm.add_constraint(x + y + x*y <= 1, label='c0')
+            'c0'
+            >>> cqm.constraints['c0'].to_polystring()
+            'x + y + x*y <= 1'
+
+        Example:
+
+            Encode a constraint with a symbolic right-hand-side.
+
+
+            .. math::
+
+                x + y \le x y
+
+
+            First create the relevant variables and the model.
+
+            >>> x, y = dimod.Binaries(['x', 'y'])
+            >>> cqm = dimod.ConstrainedQuadraticModel()
+
+            Trying to directly compare the left- and right-hand-side
+            of the equation will raise an error, because the right-hand-side
+            is not an integer or float.
+
+            >>> try:
+            ...     cqm.add_constraint(x + y <= x * y)
+            ... except TypeError:
+            ...     print("Not allowed!")
+            Not allowed!
```

Comment:
Seems a bit of an overkill -- why not just state that you cannot have the model ``x*y`` in the RHS and then say that to overcome this, do the subtraction?

Reply:
Honestly so the doctest picks it up. A little bit of future proofing honestly.

Reply:
Okay, FYI there are options to hide doctest tests in the built HTML, so you can use that for future proofing